tests/js/session-tests.js

```diff
@@ -929,5 +929,31 @@ module.exports = {
             });
             realm.close();
         });
+    },
+
+    async testAnalyticsSubmission() {
+        const context = node_require('realm/package.json');
+        const analytics = node_require('realm/lib/submit-analytics');
+
+        const payload = await analytics.fetchPlatformData(context, 'TestEvent')
+        .catch((e) => {
+            return Promise.reject("Failed to fetch platform data  [Error: " + e + "]");
+        });
+
+        TestCase.assertDefined(payload.webHook);
+        TestCase.assertType(payload.webHook.event, 'string');
+        TestCase.assertDefined(payload.webHook.properties);
+        TestCase.assertType(payload.webHook.properties.Binding, 'string');
+        TestCase.assertDefined(payload.mixPanel);
+        TestCase.assertType(payload.mixPanel.event, 'string');
+        TestCase.assertDefined(payload.mixPanel.properties);
+        TestCase.assertType(payload.mixPanel.properties.Binding, 'string');
+
+        await analytics.submitStageAnalytics('TestEvent')
+        .catch((e) => {
+            return Promise.reject("Failed to submit webhook analytics  [" + e + "]")
+        });
+
+        return Promise.resolve();
```

Comment:
No need for this - the function is `async` so it always implicitly returns a resolved promise of `undefined` when it reaches this point of execution.
```suggestion
```